asas1asas200/NTOUClassroomBorrowing
# DB connection with user login/register and CI

## Types of changes



## Description

Database connection has been done in this PR.
I use docker to create a local mongodb since it is more simply and convenient, please follow the README to setup your environment.

And the basic user login/register functions also have been done.

Also add basic test functions for CI and selftest.

## Steps to Test This Pull Request

1. Setup the docker container as README.

2. Run `npm start`.
If no problems, the console will show as below:
`MongoDB database connection established successfully`

3. Go to these pages
http://localhost:3000/users/logined
http://localhost:3000/users/admin_only
You will be redirected to http://localhost:3000/users/login

4. Go to http://localhost:3000/users/login
Login with typing `root` as id and password.
Try to access pages above, you will see some messages.

5. Go to http://localhost:3000/users/new
Register a new user then login.

6. Try to access pages above, you can only visit http://localhost:3000/users/logined .

## Related Issue
close #14 
close #32 
close #30


**Additional context**

There are still some todos in my code.
